### PR TITLE
Step 5. Create project specification

### PR DESCRIPTION
**Pull request by willyfox07**

### CODE REVIEW

**Comment by willyfox07** [comment]
я тут с коммитами переборщил) но вроде уже понял как делать так чтобы их не было так много

File: README.md [hunk 1]
```diff
@@ -1,2 +1,228 @@
...
+
+![Add employee](/image/add_employee.png)
+
+**When adding a employee, the following details are entered:**
+
+- Employee name;
+- Employee surname;
+- Hiring date - date when employee was hired;
+- Position held - what position employee does;
+- Department - which department does the employee belong to.
```
Comment (inauros) [inline README.md:71]: The department field must be a dropdown, not text - you must allow associating employees to an existing (in the database) department.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,228 @@
...
+
+### 1.3 Edit employee
+
+**Main scenario:**
+
+- User clicks the "Edit" button in the employee list view mode;
+- Application displays form to enter employee data;
+- User enters employee data and presses "Cancel" button;
```
Comment (inauros) [inline README.md:79]: I suppose, user should update, not enter the data

File: README.md [hunk 3]
```diff
@@ -1,2 +1,228 @@
...
+- Data don't save in data base, then list of employees records is displaying to user;
+- If the user select the menu item "Employees", "Departments", the data will not be saved to the database and the corresponding form with updated data will be opened.
+
```
Comment (inauros) [inline README.md:82]: Wrong main scenario

File: README.md [hunk 4]
```diff
@@ -1,2 +1,228 @@
...
+**Cancel operation scenario:**
+
+- User clicks the “Edit” button in the employee list view mode;
+- Application displays form to enter employee data;
+- User enters employee data and presses “Cancel” button;
+- Data don’t save in data base, then list of employee records is displaying to user.
+- If the user selects the menu item "Employees”, ”Departments”, the data will not be saved to the database and the corresponding form with updated data will be opened.
+
+![Edit employee](/image/edit_employee.png)
```
Comment (inauros) [inline README.md:91]: For the edit scenario should be used another mockup (data should be pre-filled)

File: README.md [hunk 5]
```diff
@@ -1,2 +1,228 @@
...
+- Hiring date - format dd/mm/yyyy;
+- Position held - maximum length of 90 characters;
+- Department - maximum length of 90 characters.
+
+### 1.4 Removing employee
+
+**Main scenario:**
+
+- The user, while in the list of employees, presses the "Delete" button in the selected employee line;
+- If the employee can be removed, a confirmation dialog is displayed;
```
Comment (inauros) [inline README.md:114]: Why "If"? Are there cases, when the employee can't be removed? If yes, please could you list them?

File: README.md [hunk 6]
```diff
@@ -1,2 +1,228 @@
...
+- The user confirms the removal the employee;
+- Record is delected from database;
+- If error occurs, then error message displays;
+- If employee data is succesfully deleted, then list of employee without delected data is displaying. 
+
+![Edit employee](/image/delete_order.png)
```
Comment (inauros) [inline README.md:120]: As I can see on mockup, you are trying to remove order, not employee ;)

File: README.md [hunk 7]
```diff
@@ -1,2 +1,228 @@
...
+- Application displays list of employees;
+
+![Edit employee](/image/Departments.png)
+
+**The list displays the following columns:**
+
+- Department - department name;
+- The number of employees - how mane employees in this dapetment;
+- Head - head of department;
+- Location - where this department located.
```
Comment (inauros) [inline README.md:147]: the average salary column is missed

File: README.md [hunk 8]
```diff
@@ -1,2 +1,228 @@
...
+## Departments
+
+### 2.1 Display list of Departments 
+
+This mode is intended for viewing and editing the departments list.
+
+**Main scenario:**
+
+- User selects item "Departments";
+- Application displays list of employees;
```
Comment (inauros) [inline README.md:138]: I suppose departments not employees

File: README.md [hunk 9]
```diff
@@ -1,2 +1,228 @@
...
+![Edit employee](/image/Departments.png)
+
+**The list displays the following columns:**
+
+- Department - department name;
+- The number of employees - how mane employees in this dapetment;
+- Head - head of department;
+- Location - where this department located.
+
+**Filtering by location:**
```
Comment (inauros) [inline README.md:149]: 1. I guess other fields should also be involved in filtering
2. how about sorting?

File: README.md [hunk 10]
```diff
@@ -1,2 +1,228 @@
...
+### 2.3 Edit department
+
+**Main scenario:**
+
+- User clicks the "Edit" button in the department list view mode;
+- Application displays form to enter department data;
+- User enters department data and presses "Cancel" button;
+- Data don't save in data base, then list of department records is displaying to user;
+- If the user select the menu item "Employees", "Departments", the data will not be saved to the database and the corresponding form with updated data will be opened.
+
```
Comment (inauros) [inline README.md:199]: The same comments as for employees.

File: README.md [hunk 11]
```diff
@@ -1,2 +1,228 @@
...
+**Cancel operation scenario:**
+
+- User clicks the “Edit” button in the department list view mode;
+- Application displays form to enter department data;
+- User enters department data and presses “Cancel” button;
+- Data don’t save in data base, then list of department records is displaying to user.
+- If the user selects the menu item "Employees”, ”Departments”, the data will not be saved to the database and the corresponding form with updated data will be opened.
+
+![Edit department](/image/edit_department.png)
```
Comment (inauros) [inline README.md:208]: The same as for employees - please fix the mockup

File: README.md [hunk 12]
```diff
@@ -1,2 +1,228 @@
...
+**Main scenario:**
+
+- The user, while in the list of department, presses the "Delete" button in the selected department line;
+- If the department can be removed, a confirmation dialog is displayed;
+- The user confirms the removal the department;
+- Record is delected from database;
+- If error occurs, then error message displays;
+- If department data is succesfully deleted, then list of department without delected data is displaying.
+
+![Delete department](/image/delete_department.png)
```
Comment (inauros) [inline README.md:221]: Again, you are removing Order not department :)

**Review by inauros — CHANGES_REQUESTED** [review]
1. Please, when you create PRs, name them in accordance with the current step.
2. In accordance with step 4 in Grow, you had to include a README file with details about the project goal in English.
2. In accordance with step 5 in Grow, you had to create a folder "documentation" at the root of your repository - this folder must include all files related to SRS
3. Date of Birth should be used instead of Hiring Date
4. For all negative cases (employee/department can't be updated/saved/removed, etc.), mockups also should be created.
5. please remove pythonProject file

File: README.md [hunk 13]
```diff
@@ -1,2 +1,228 @@
-# pythonProject
+# Employee Manager
 
+## Vision
+
+"Employee Manager" s a web application designed for the management of employees and departments with these employees.
```
[the remark this answers is not in the document]
Comment (willyfox07) [reply README.md:5]: fixed

File: README.md [hunk 14]
```diff
@@ -1,2 +1,228 @@
...
+
+Applecation should provide:
```
[the remark this answers is not in the document]
Comment (willyfox07) [reply README.md:7]: fixed

File: README.md [hunk 15]
```diff
@@ -1,2 +1,228 @@
...
+
+- Storing information about employees in a database;
+- Display list of employees;
+- Updating list of employees(adding,editing,removing);
+- Display list of departments;
+- Updating list of depatrments;
```
[the remark this answers is not in the document]
Comment (willyfox07) [reply README.md:13]: fixed

File: README.md [hunk 16]
```diff
@@ -1,2 +1,228 @@
...
+- Filtering by departments for employees;
+- Filtering by hiring date for employees;
```
[the remark this answers is not in the document]
Comment (willyfox07) [reply README.md:15]: fixed

File: README.md [hunk 17]
```diff
@@ -1,2 +1,228 @@
...
+
+The mode is designed to view the list of employees.
+
+**Main scenario**
+
+- User select item "Employees";
+- Application displays list of Employees.
+
+![Main menu](/image/Start.png)
+Pic.1.1 View the list of employees.
```
[the remark this answers is not in the document]
Comment (willyfox07) [reply README.md:29]: fixed

File: README.md (hunk 9 above)
Comment (willyfox07) [reply README.md:149]: fixed

File: README.md (hunk 11 above)
Comment (willyfox07) [reply README.md:208]: fixed

File: README.md (hunk 12 above)
Comment (willyfox07) [reply README.md:221]: fixed

File: README.md (hunk 10 above)
Comment (willyfox07) [reply README.md]: Тут я не очень понимаю что мне нужно исправить)

File: README.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (inauros) [reply README.md]: I still don't see the Salary column )

[the remark this answers is not in the document]
Comment (willyfox07) [reply README.md]: fixed

File: README.md [hunk 18]
```diff
@@ -1,2 +1,228 @@
...
+- Data don’t save in data base, then list of employees records is displaying to user.
+- If the user selects the menu item "EMployees”, ”Departments”, the data will not be saved to the database and the corresponding form with updated data will be opened.
+
+![Add employee](/image/add_employee.png)
+
+**When adding a employee, the following details are entered:**
+
+- Employee name;
+- Employee surname;
+- Hiring date - date when employee was hired;
```
[the remark this answers is not in the document]
Comment (willyfox07) [reply README.md]: fixed

File: README.md (hunk 4 above)
Comment (willyfox07) [reply README.md]: fixed

File: README.md (hunk 1 above)
Comment (willyfox07) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (willyfox07) [reply README.md]: fixed

File: README.md (hunk 6 above)
Comment (willyfox07) [reply README.md]: fixed

File: README.md (hunk 5 above)
Comment (willyfox07) [reply README.md]: fixed

File: README.md (hunk 7 above)
Comment (willyfox07) [reply README.md]: fixed

File: README.md (hunk 10 above)
Comment (willyfox07) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (willyfox07) [reply README.md]: fixed

File: README.md (hunk 7 above)
Comment (willyfox07) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (willyfox07) [reply README.md]: fixed

File: README.md [hunk 19]
```diff
@@ -1,2 +1,228 @@
...
+- Name - Name of employees;
+- Surname - Surname of employees;
+- Hiring date - date when employee was hired;
+- Position held - what position employee does;
+- Department - which department does the employee belong to.
+
+**Filtering by date:**
+
+- In the list of employees mode, the user sets a date filter and presses the refresh list button
+
```
[the remark this answers is not in the document]
Comment (willyfox07) [reply README.md]: fixed

**Review by inauros — APPROVED** [review]

**Pull request merged**